crates/starknet/src/starknet/mod.rs

```diff
@@ -728,6 +728,8 @@ impl Starknet {
             )?;
 
             let state_diff: ThinStateDiff = state.extract_state_diff_from_pending_state()?.into();
+            let state_diff =
+                if state_diff == ThinStateDiff::default() { None } else { Some(state_diff) };
```

Comment:
This looks like a hack we might forget about. Why doesn't `extract_state_diff_from_pending_state` return `Option`?

Reply:
Return data of `extract_state_diff_from_pending_state` is used as a required field for `starknet_getStateUpdate` rpc method, then in that case an option cant be returned. Also checking the `CommitmentStateDiff` in `extract_state_diff_from_pending_state` requires checking every property of the struct if its empty, because `CommitmentStateDiff` doesnt have default method implemented.

Reply:
I'll resolve this conversation then